backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/AbstractDunningDAO.java

```diff
@@ -229,10 +234,26 @@ public final List<I_C_DunningDoc_Line_Source> retrieveDunningDocLineSources(@Non
 
 	protected abstract Iterator<I_C_Dunning_Candidate> retrieveDunningCandidatesIterator(IDunningContext context, IDunningCandidateQuery query);
 
-
 	@Override
 	public I_C_DunningDoc getByIdInTrx(@NonNull final DunningDocId dunningDocId)
 	{
 		return load(dunningDocId, I_C_DunningDoc.class);
 	}
+
+	@Override
+	public Collection<I_C_DunningDoc> getByIdsInTrx(@NonNull final Collection<DunningDocId> dunningDocIds)
+	{
+		return queryBL.createQueryBuilder(I_C_DunningDoc.class)
+				.addInArrayFilter(I_C_DunningDoc.COLUMNNAME_C_DunningDoc_ID, dunningDocIds)
+				.addOnlyActiveRecordsFilter()
+				.create()
+				.list();
+	}
+
+	@Override
+	@Cached
+	public String getDunningLevelName(final @NonNull DunningLevelId id)
```

Comment:
avoid `@Cached`

Reply:
consider introducing a minimal DunningLevel and a DunningLevelMap which contains all